include/openssl/ec_key.h

```diff
@@ -356,6 +328,115 @@ OPENSSL_EXPORT EC_KEY *o2i_ECPublicKey(EC_KEY **out_key, const uint8_t **inp,
 OPENSSL_EXPORT int i2o_ECPublicKey(const EC_KEY *key, unsigned char **outp);
 
 
+// EC_KEY_METHOD
+
+// ECDSA_FLAG_OPAQUE specifies that this EC_KEY_METHOD does not expose its key
+// material. This may be set if, for instance, it is wrapping some other crypto
+// API, like a platform key store. Use |EC_KEY_METHOD_set_flag| to set
+// this flag on an |EC_KEY_METHOD|.
+// This was supported in ECDSA_METHOD previously and is not set by default.
+#define ECDSA_FLAG_OPAQUE 1
+
+// EC_KEY_get_default_method returns a reference to the default
+// |EC_KEY| implementation. All |EC_KEY| objects are initialized with the
+// returned struct. This function currently calls |EC_KEY_OpenSSL| since AWS-LC
+// does not support changing/setting the default method.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_get_default_method(void);
+
+// EC_KEY_OpenSSL returns a reference to the default |EC_KEY| implementation.
+// The returned |EC_KEY_METHOD| object is statically allocated. The application
+// should not free this struct.
+//
+// This struct is also zero-initialized. This is different from OpenSSL which
+// returns function pointers to the default implementations within the
+// |EC_KEY_METHOD| struct. We do not do this to make it easier for the
+// compiler/linker to drop unused functions. The wrapper functions will select
+// the appropriate |ec_key_default_*| implementation.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_OpenSSL(void);
+
+// EC_KEY_METHOD_new returns a newly allocated |EC_KEY_METHOD| object. If the
+// input parameter |eckey_meth| is non-NULL, the function pointers within the
+// returned |EC_KEY_METHOD| object will be initialized to the values from
+// |eckey_meth|. If |eckey_meth| is NULL, the returned object will be
+// initialized using the value returned from |EC_KEY_get_default_method|.
+OPENSSL_EXPORT EC_KEY_METHOD *EC_KEY_METHOD_new(const EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_METHOD_free frees the memory associated with |eckey_meth|
+OPENSSL_EXPORT void EC_KEY_METHOD_free(EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_set_method sets |meth| on |ec|. While we provide setters to set
+// the |copy|, |set_group|, |set_private|, |set_public|, and |sign_setup|
+// fields, we do not support the associated functionality and these pointers
+// should be set to NULL. We do not support the |verify|, |verify_sig|, or
+// |keygen| fields yet.
+//
+// Returns zero on failure and one on success.
+OPENSSL_EXPORT int EC_KEY_set_method(EC_KEY *ec, const EC_KEY_METHOD *meth);
+
+// EC_KEY_get_method returns the |EC_KEY_METHOD| object associated with |ec|.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_get_method(const EC_KEY *ec);
+
+#ifdef __cplusplus
+#define IS_NULL(ptr) ((ptr) == nullptr)
+#else
+#define IS_NULL(ptr) ((ptr) == NULL)
+#endif
```

Comment:
* This name `IS_NULL` seems likely to conflict with a macro defined by a consumer. What about `AWSLC_IS_NULL`?
* Why not just use `OPENSSL_STATIC_ASSERT` in the places it's needed below. Although discouraged, I believe the `NULL` macro is still available with C++.

Reply:
Got it, changed to use OPENSSL_STATIC_ASSERT to support C99. Got rid of the IS_NULL macro, NULL does exist in C++ and should be okay for consumers to use.